docs/variant_merging.md

```diff
@@ -1,3 +1,227 @@
 # Variant merging
 
-TODO
+You can use Variant Transforms to load multiple files at once
+by providing a pattern like `--input_pattern gs://mybucket/**.vcf`. This
+will create a merged BigQuery schema that contains all variants from all
+matching files. Note that the VCF fields must be compatible across files (e.g.
+`INFO` and `FORMAT` fields with the same key have compatible types and numbers
+across files).
+
+However, by default, calls between files will not be merged. For instance,
+consider the following VCF files:
+
+*`gs://mybucket/a.vcf`*:
+
+```
+## <headers omitted>
+#CHROM  POS   ID    REF   ALT   QUAL  FILTER  INFO    FORMAT  S1      S2
+1       100   rs1   A     T     29    PASS    DP=14   GT:GQ   0|1:48  0/0:30
+```
+
+*`gs://mybucket/b.vcf`*:
+
+```
+## <headers omitted>
+#CHROM  POS   ID    REF   ALT   QUAL  FILTER  INFO         FORMAT  S3
+1       100   rs1   A     T     7     q10     DP=11;AA=G   GT:GQ   1/1:2
+```
+
+If you load both files together (i.e. by specifying
+`--input_pattern gs://mybucket/*.vcf`), you would end up with 2 rows in
+BigQuery. One with calls `S1` and `S2`, and another with just call `S3` even
+though all calls have the same variant.
+
+Variant merging can be used to merge calls across files. There are different
+merging strategies and you can specify any of the following
+strategies by adding `--variant_merge_strategy <merge_strategy>` to the
+arguments.
+
+## MOVE_TO_CALLS strategy
+
+This strategy merges calls that have the same `reference_name`,
+`start_position`, `end_position`, `reference_bases`, and all `alternate_bases`
+across files. By default, the variant fields are merged as follows:
+
+* `Calls` are merged together in no particular order.
+* The highest `quality` value is chosen among all variants being merged.
+* All `filter` values are merged and deduplicated.
+* All `names` (`ID` column in the VCF spec) are merged and deduplicated.
+* For each `INFO` field, one is chosen as representative for the *entire* merged
+  variant record in *no particular order*.
+
+Depending on your use case, you may customize the merge strategy as follows:
+
+* Adding `--copy_quality_to_calls` and/or `--copy_filter_to_calls` to the
+  arguments will copy the `quality` and/or `filter` values in each file to the
+  set of calls specified in that file. This option is useful when merging
+  single-call VCF files as `quality` and `filter` values usually correspond
+  the call-level quality/filter. Note that the variant-level `quality` and
+  `filter` values will still be merged according to the above logic.
+* Adding `--info_keys_to_move_to_calls_regex <regex>` will move `INFO` fields
+  that match the provided regular expression to the associated calls. This is
+  useful to ensure all `INFO` fields are kept when merging variants.
```

Comment:
What happens in FORMAT of two calls are different? Should you document that too?

Reply:
Good point, but I think this is independent of the merging logic, as it can happen when loading multiple files without merging. I'm planning to write a doc on 'importing multiple files' and will explain this as well.

Reply:
Okay makes sense.